### PR TITLE
[ENH] Add ability to draw graphs with a title

### LINKED ISSUE #23: Add ability to add title to viz.draw
**Is your feature request related to a problem? Please describe.**
I want to visualize several graphs and distinguish them with a title.

**Describe the solution you'd like**
An optional argument for an image title.

**Describe alternatives you've considered**
Adding title outside of image.  A bit clumsy.

**Additional context**
I'm trying to compare graphs.

### PR DESCRIPTION
**Pull request by aryan26roy**
Closes #23 

Changes proposed in this pull request:

- Adds a keyword argument for graph labels to viz.draw

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Review by adam2392 — COMMENTED** [review]
Can you add a unit-test and also add a changelog entry to `docs/whats_new/v0.1.rst`?

**Comment by aryan26roy** [comment]
Sure! Just getting used to the procedure for this repo.

File: docs/whats_new/v0.1.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Version 0.1
 
 Changelog
 ---------
-
+- |Feature| Add keyword argument for graph lables in :func:`pywhy_graphs.viz.draw`, by `Aryan Roy`_ (:pr:`71`)
```
Comment (adam2392) [inline docs/whats_new/v0.1.rst:28]:
```suggestion
- |Feature| Add keyword argument for graph labels in :func:`pywhy_graphs.viz.draw`, by `Aryan Roy`_ (:pr:`71`)
```

**Review by adam2392 — CHANGES_REQUESTED** [review]
Thanks for the PR! It mostly looks good minus a few nitpicks. 

One last thing: Can you also augment the existing example in `examples/` to show off the label feature? E.g. in https://output.circle-artifacts.com/output/job/e94330e9-bd6c-4d83-ae14-8ed903121326/artifacts/0/dev/auto_examples/visualization/draw_and_compare_graphs_with_same_layout.html#sphx-glr-auto-examples-visualization-draw-and-compare-graphs-with-same-layout-py it would be great to see some sensible titles of the different graphs.

You can build the documentation locally to test it out and what it looks like by following the Contributing.md doc. Lmk if you have any issues!

File: pywhy_graphs/viz/tests/test_draw.py [hunk 2]
```diff
@@ -105,3 +105,14 @@ def test_draw_with_ts_layout():
...
+def test_draw_name_is_given():
+    """
+    Ensure the generated graph contains the lable provided by the user.
+    """
+    # create a dummy graph x --> y <-- z and z --> x
+    graph = nx.DiGraph([("x", "y"), ("z", "y"), ("z", "x")])
+    # draw the graphs
+    dot = draw(graph, name="test")
+    # assert that the produced graph contains the label for the graph
+    assert "label=test" in dot.source
```
[the remark this answers is not in the document]
Comment (aryan26roy) [reply pywhy_graphs/viz/tests/test_draw.py]: Sure!

**Comment by aryan26roy** [comment]
Hey @adam2392 I think I implemented all the changes you requested. Sorry for the delay though.

**Review by adam2392 — APPROVED** [review]
LGTM once CI passes

**Comment by adam2392** [comment]
Thanks @aryan26roy !

**Pull request merged**